apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/DatasetsTest.cs

```diff
@@ -145,5 +146,120 @@ public void PatchDataset_ConflictDontMatchEtag()
             Assert.Equal("Description2", fetched.Resource.Description);
             Assert.Equal("Sneak attack!", fetched.Resource.FriendlyName);
         }
+
+        [Fact]
+        public async Task UpdateDatasetAsync()
+        {
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            var id = _fixture.CreateDatasetId();
+
+            var original = await client.CreateDatasetAsync(id, new CreateDatasetOptions { Description = "Description1", FriendlyName = "FriendlyName1" });
+
+            // Modify locally...
+            original.Resource.Description = "Description2";
+            original.Resource.FriendlyName = "FriendlyName2";
+
+            // FIXME: I shouldn't need to do this.
+            original.Resource.ETag = client.GetDataset(id).Resource.ETag;
```

Comment:
I left that deliberately as:

- Having synchronous operations within an async test isn't really a problem; I just *generally* try to make them consistent
- In this case it would make the code uglier with parentheses: `(await client.GetDatasetAsync(id)).Resource.ETag;`
- I want to be able to remove that line of code entirely :)